src/linker/Linker.Steps/ResolveFromXmlStep.cs

```diff
@@ -86,17 +89,30 @@ protected override void Process ()
 				if (Context.IgnoreDescriptors)
 					return;
 			}
+
+#if FEATURE_ILLINK
```

Comment:
Could you if-def the implementation instead of many `#if FEATURE_ILLINK` for ShouldProcessElement